### PR TITLE
Task 1.2 & 2.1 solved

### PR DESCRIPTION
**Pull request by MaximKuznetsov17**
Реализован аналитический и численный расчёт градиента, softmax и cross-entropy, L2-регуляризацию, натренирован линейный классификатор. Точность составила 0.159. 

Извините за опоздание, не рассчитал время:(

### CODE REVIEW

**Comment by MaximKuznetsov17** [comment]
Добавил решение таски 2.1
Пока что решение получилось крайне плохим, попробую оперативно разобраться в чём проблема. Но добавил в сейчас это решение в ПР из-за наступления дедлайна
Neural net test set accuracy: 0.182000 
Loss: 2.267202, Train accuracy: 0.196667, val accuracy: 0.206000

**Comment by toshiks** [comment]
Поставим за 1.2 0.3*8 баллов

**Comment by MaximKuznetsov17** [comment]
> Поставим за 1.2 0.3*8 баллов

@toshiks 
А можно объяснить как эта система работает?
